src/include/firebird/FirebirdInterface.idl

```diff
@@ -1487,3 +1507,51 @@ interface DecFloat34 : Versioned
 	void fromBcd(int sign, const uchar* bcd, int exp, FB_DEC34* to);
 	void fromString(Status status, const string from, FB_DEC34* to);
 }
+
+// Replication interfaces
+
+interface ReplicatedRecord : Versioned
+{
+	uint getRawLength();
+	const uchar* getRawData();
+}
+
+interface ReplicatedBlob : Versioned
+{
+	uint getLength();
+	boolean isEof();
+	uint getSegment(uint length, uchar* buffer);
+}
+
+interface ReplicatedTransaction : Disposable
+{
+	boolean prepare();
+	boolean commit();
+	boolean rollback();
+
+	boolean startSavepoint();
+	boolean releaseSavepoint();
+	boolean rollbackSavepoint();
+
+	boolean insertRecord(const string name,
+					  	 ReplicatedRecord record);
+	boolean updateRecord(const string name,
+					  	 ReplicatedRecord orgRecord,
+					  	 ReplicatedRecord newRecord);
+	boolean deleteRecord(const string name,
+					  	 ReplicatedRecord record);
+
+	boolean storeBlob(ISC_QUAD blobId, ReplicatedBlob blob);
+
+	boolean executeSql(const string sql);
+}
+
+interface ReplicatedSession : Disposable
+{
+	Status getStatus();
+
+	ReplicatedTransaction startTransaction(int64 number);
+	boolean cleanupTransaction(int64 number);
```

Comment:
Some comments about "what this method is supposed to do and when is called" would be very appreciated.